vectorstores/alloydb/engine.go

```diff
@@ -50,24 +44,35 @@ func NewPostgresEngine(ctx context.Context, opts ...Option) (*PostgresEngine, er
 }
 
 // createConnection creates a connection pool to the PostgreSQL database.
-func createConnection(ctx context.Context, cfg engineConfig) (*pgxpool.Pool, error) {
-	// Create a new dialer with any options
-	d, err := alloydbconn.NewDialer(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize connection: %w", err)
+func createConnection(ctx context.Context, cfg engineConfig, usingIAMAuth bool) (*pgxpool.Pool, error) {
+	var d *alloydbconn.Dialer
+	var dsn string
+	var err error
+
+	// Create a new dialer to connect to the database.
+	if !usingIAMAuth {
+		d, err = alloydbconn.NewDialer(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize connection: %w", err)
+		}
+		dsn = fmt.Sprintf("user=%s password=%s dbname=%s sslmode=disable", cfg.user, cfg.password, cfg.database)
+	} else {
+		d, err = alloydbconn.NewDialer(ctx, alloydbconn.WithIAMAuthN())
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize connection: %w", err)
+		}
+		dsn = fmt.Sprintf("user=%s dbname=%s sslmode=disable", cfg.user, cfg.database)
 	}
```

Comment:
Suggest you simplify this to give you self more wiggle room in the future in case more options become used:

```suggestion
	dsn := fmt.Sprintf("user=%s password=%s dbname=%s sslmode=disable", cfg.user, cfg.password, cfg.database)
	dialeropts := alloydbconn.DialerOptions[]{}
	
	if usingIAMAuth {
	 	dialeropts = append(dialeropts, alloydbconn.WithIAMAuthN())
		dsn = fmt.Sprintf("user=%s dbname=%s sslmode=disable", cfg.user, cfg.database)
	}

	d, err = alloydbconn.NewDialer(ctx)
	if err != nil {
		return nil, fmt.Errorf("failed to initialize connection: %w", err)
	}

```